# Práctica 2
### Computación
Grupo 8093

Prof. Sergio Antonio Alcalá Corona

Ernesto Andreo Chimal García

---

### *cat,  head, tail*

In [3]:
mkdir Resultado

mkdir: Resultado: File exists


: 1

Creé el directorio *Resultado*.

In [5]:
cp Computacion2020-8093/Practicas/directorio/TLALPAN/sectorH/names_2009-08-30.txt Resultado

Copié el directorio */directorio/TLALPAN/sectorH/names_2009-08-30.txt* en el directorio *Resultado*.

In [7]:
cd Resultado

Me trasladé al directorio *Resultado*.

In [ ]:
cat < names_2009-08-30.txt > names_cat.txt

In [9]:
cat names_cat.txt

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	5291161226

Alba Anabel	5642845849

Aguilar Alexia	5462563799

Serralde Angel Sebastian	5522588882

Yines Ana Ximena	5593340690

Calette Alexia	5032726714

Balmaceda Jose De	5214401999

Cerezo Jesica Jocelyn	5003217550

Garnica Kevin	5978399615



Ejecuté el comando *cat < names_2009-08-30.txt > names_cat.txt*.

El comando *cat* leyó los contenidos del archivo de texto *names_2009-08-30.txt*, y redirigió el output al archivo *names_cat.txt* (que se creó con este comando, pues no existía).

In [10]:
cat < names_2009-08-30.txt >> names_cat.txt

In [11]:
cat names_cat.txt

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	5291161226

Alba Anabel	5642845849

Aguilar Alexia	5462563799

Serralde Angel Sebastian	5522588882

Yines Ana Ximena	5593340690

Calette Alexia	5032726714

Balmaceda Jose De	5214401999

Cerezo Jesica Jocelyn	5003217550

Garnica Kevin	5978399615

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	52

Ejecuté el comando *cat < names_2009-08-30.txt >> names_cat.txt*.

El comando *cat* leyó los contenidos del archivo de texto *names_2009-08-30.txt*, y redirigió el output al archivo *names_cat.txt*; este redireccionamiento fue no destructivo, por lo que ahora en el archivo *names_cat.txt* aparece dos veces seguidas el contenido de *names_2009-08-30.txt*.

In [157]:
cat names_cat.txt 2> err.dat

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	5291161226

Alba Anabel	5642845849

Aguilar Alexia	5462563799

Serralde Angel Sebastian	5522588882

Yines Ana Ximena	5593340690

Calette Alexia	5032726714

Balmaceda Jose De	5214401999

Cerezo Jesica Jocelyn	5003217550

Garnica Kevin	5978399615

Aguilar Natasha Alejanda	5908405556

Salomon Valeria	5550562274

Cedillo Arath Abijah	5957438808

Lazaro Heriberto	5131681455

Muñoz Maria Renee	5289217179

Saldaña Francisco	5393306394

Ramos Carmen Sofía	5564653521

Buneder Erick Ricardo	5853411637

Balandra Ana Ximena	5203278041

Molina Eduardo Alberto	5736392724

Barranco Carol Sthefany	5664011544

Santiago Lesli Paola	52

In [159]:
cata names_cat.txt 2> err.dat

: 127

In [160]:
cat err.dat

bash: cata: command not found


La diferencia entre ambos comandos es que *cata* no existe en *PATH* (este comando no existe en mi computadora). El primer comando leyó los contenidos de *names_cat.txt* y creó el archivo *err.dat*, pero no los redirigió a *err.dat*, sino que los imprimió en pantalla y dejó vacío el archivo. El segundo comando, al no existir, dejó un pequeño texto de error en el archivo *err.dat* (el texto dice "bash: cata: command not found").

In [64]:
cat feliz

Hola, buenas tardes.





Creé un archivo de texto llamado *feliz*, dentro del cual se encuentra el texto
"Hola, buenas tardes."

Los espacios en blanco se deben a que di enter antes de ejecutar el segundo *Ctrl+D*.

In [71]:
wc -l ../Computacion2020-8093/Practicas/directorio/COYOACAN/sectorG/names_2005-09-27.dat

      40 ../Computacion2020-8093/Practicas/directorio/COYOACAN/sectorG/names_2005-09-27.dat


In [73]:
tail -40r ../Computacion2020-8093/Practicas/directorio/COYOACAN/sectorG/names_2005-09-27.dat > catinverso.dat

Primero conté la cantidad de líneas en el archivo *names_2005-09-27.dat*, y después ejecuté el comando tail (que imprime las últimas $n$  líneas de un archivo de texto) con la opción *-40* para imprimir las 40 líneas del archivo y la opción *-r* para imprimirlas en reversa.

In [139]:
cd /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio/CUAUHTEMOC/sectorE

In [142]:
cat * > /Users/andreochimal/Desktop/Compu/Resultado/todosarchivos.txt

Me trasladé al directorio */Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio/CUAUHTEMOC/sectorE* y ejecuté el comando *cat* con un comodín, el cual concatenó los contenidos de todos los archivos que cumplieran con el comodín (que son todos, pues el comodín no trae especificación alguna). Eso lo redirigí al archivo *todosarchivos.txt*, que recién creé en la carpeta *Resultado*.

In [143]:
cd /Users/andreochimal/Desktop/Compu/Resultado

In [144]:
cat todosarchivos.txt

Razo Pedro	5078328708

Zavala Oscar Enrique	5349803565

Arevalo Anabel	5098283547

Castillo Regina	5585642291

Muñoz Alexia	5670317235

Molina Montserrat	5506646832

Paredes Jose Manuel	5683845245

Meza Jovanna Sofia	5197111714

Hernandez Ana Karen	5000717497

Cejudo Elsy Daniela	5571564174

Sarabia Ana Ximena	5563916832

Cerezo Santiago Miguel	5018574468

Mendez Abiram	5908264097

Vidaños Jose De	5595988687

Cedillo Jesica Jocelyn	5948863480

Romero Alexis	5584398354

Sierra Heriberto	5497822462

Contreras Logan Alexander	5021410239

Munguia Luis Fernando	5120254168

Orozco María Fernanda	5469440435

Rosales Maroon Liliana	5459481540

Yines Aimee Berenice	5594659911

Marquez Kevin Kaled	5322379413

Caballero Jose De	5866484113

Solís Jesus	5954572200

Lagunas Gerardo	5543595417

Fernandez Will Cameron	5410185674

Saldaña Mariana Camila	5767575914

Mendez Emmanuel	5263539085

Corona Ana Paola	5225467712

Piña Francisco Alexander	5028397432

Silva Aimee Berenice	5381719174

Manriquez Ma

Me trasladé a la carpeta *Resultado* y comprobé que sí se imprimieran correctamente los contenidos de *../Computacion2020-8093/Practicas/directorio/CUAUHTEMOC/sectorE* en el archivo de texto *todosarchivos.txt*.

In [154]:
cat -n todosarchivos.txt | head -23 > colacabeza; cat -n todosarchivos.txt | tail -32 >> colacabeza

In [155]:
cat colacabeza

     1	Razo Pedro	5078328708
     2	
     3	Zavala Oscar Enrique	5349803565
     4	
     5	Arevalo Anabel	5098283547
     6	
     7	Castillo Regina	5585642291
     8	
     9	Muñoz Alexia	5670317235
    10	
    11	Molina Montserrat	5506646832
    12	
    13	Paredes Jose Manuel	5683845245
    14	
    15	Meza Jovanna Sofia	5197111714
    16	
    17	Hernandez Ana Karen	5000717497
    18	
    19	Cejudo Elsy Daniela	5571564174
    20	
    21	Sarabia Ana Ximena	5563916832
    22	
    23	Cerezo Santiago Miguel	5018574468
   769	Maya Ximena Itzel	5187140592
   770	
   771	Cazares Heriberto	5591260956
   772	
   773	Saldaña Katia Mariana	5297695281
   774	
   775	Balandra Mabel	5346871896
   776	
   777	Aguilar Maria Fernanda	5620980938
   778	
   779	Cazares Ximena	5031790603
   780	
   781	Nieto Oscar Enrique	5529321728
   782	
   783	Ruiz Alexis	5330479468
   784	
   785	Serrano Luis Fernando	5429722585
   786	
   787	Avila Monserrat	5896435866
   788	
   789	Ramos Martha	5849299706
   790	
 

En una sola celda hice lo siguiente:

Ejecuté el comando *cat -n* para que se imprimieran numeradas todas las líneas de *todosarchivos.txt*, y eso lo redirigí al comando *head -23*, que imprimió solamente las primeras 23 líneas, y eso lo redirigí al archivo colacabeza (que se creó en este comando, pues no existía).

Después hice algo muy similar, pero redirigí lo que imprimió el comando *cat -n* al comando *tail -32* para imprimir las últimas 32 líneas numeradas. Esto lo redirigí de forma no destructiva a *colacabeza*.

Tras ello corroboré en el archivo *colacabeza* que sí haya ocurrido lo planeado.

### Permisos

In [1]:
umask

0022


In [4]:
cd Resultado

In [5]:
ls -la

total 80
drwxr-xr-x@ 11 andreochimal  staff    352 Nov  4 00:15 .
drwxr-xr-x@ 13 andreochimal  staff    416 Nov  5 15:59 ..
drwxr-xr-x@  3 andreochimal  staff     96 Nov  3 20:11 .ipynb_checkpoints
drwxr-xr-x@ 13 andreochimal  staff    416 Nov  4 00:15 CUAUHTEMOC
-rw-r--r--   1 andreochimal  staff    597 Nov  3 16:55 catinverso.dat
-rw-r--r--   1 andreochimal  staff   1239 Nov  3 21:32 colacabeza
-rw-r--r--   1 andreochimal  staff     30 Nov  3 21:50 err.dat
-rw-r--r--   1 andreochimal  staff     24 Nov  3 16:43 feliz
-rw-r--r--   1 andreochimal  staff    625 Nov  3 16:28 names_2009-08-30.txt
-rw-r--r--   1 andreochimal  staff   1250 Nov  3 16:29 names_cat.txt
-rw-r--r--   1 andreochimal  staff  12481 Nov  3 21:24 todosarchivos.txt


De acuerdo a una pequeña investigación en internet, los últimos tres dígitos que arroja *umask* son los permisos que se "restan" a un arh¡chivo o directorio al crearse. De forma predeterminada, los directorios tienen permisos 777, y el *umask* indica que se les está "restando" 022, por lo que los permisos del usuario permanecen intactos, al grupo se le quita el permiso de escritura y a los otros también (744). De forma predeterminada, los archivos tienen permisos 666, por lo que debe ocurrirles lo mismo, solamente que nadie tiene permiso de ejecución inicialmente (644).

Al final ejecuté *ls -la* para corroborar que sí ocurriera esto: todos los archivos tienen permiso 644 y los directorios tienen permiso 744.

**Ejemplo de uso**: Si creo una cantidad muy grande de archivos y no quiero usar el comando *ls* para conocer sus permisos (pues esto crearía una lista enorme), basta con hacer una simople operación matemática para saber qué permisos tienen todos por default.

In [163]:
cp -R ../Computacion2020-8093/Practicas/directorio/CUAUHTEMOC .

In [187]:
cd CUAUHTEMOC

In [188]:
chmod -R 614 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/*/*

In [190]:
ls -Rla

total 0
drwx--xr--@ 13 andreochimal  staff   416 Nov  3 22:49 .
drwxr-xr-x@ 11 andreochimal  staff   352 Nov  3 22:49 ..
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorA
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorE
drwx--xr--@ 42 andreochimal  staff  1344 Nov  3 22:49 sectorG
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorJ
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorM
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorN
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorO
drwx--xr--@ 42 andreochimal  staff  1344 Nov  3 22:49 sectorP
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorR
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorU
drwx--xr--@ 22 andreochimal  staff   704 Nov  3 22:49 sectorV

./sectorA:
total 160
drwx--xr--@ 22 andreochimal  staff  704 Nov  3 22:49 .
drwx--xr--@ 13 andreochimal  staff  416 Nov  3 22:49 ..
-rw---xr--   1 andreochimal  staff  618 Nov  3 22:49 names_2000-

Copié la carpeta *CUAUHTEMOC* al directorio *Resultado*, entré a la carpeta y cambié los permisos de forma recursiva a todos los directorios y subdirectorios dentro de *CUAUHTEMOC* tal como se pidió. Al final corroboré que se cambiaran como se deseaba.

In [11]:
chmod 740 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorG/*.txt

Cambié los permisos de todos los archivos que terminan en *.txt* dentro de *sectorG* a 740, tal como se pidió.

In [13]:
chmod 714 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorP/*.dat

In [14]:
ls -l /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorP/*.dat > permisos.dat

Cambié los permisos de todos los archivos que terminan en *.dat* dentro de *sectorP* a 714, tal como se pidió, enlisté los archivos que terminan en *.dat* con sus respectivos permisos y redirigí el output de esto a un archivo en *Resultado* llamado *permisos.dat*.

In [15]:
chmod 700 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sector{O,P}/*-05-*

In [17]:
ls -l /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sector{O,P}/*-05-*

-rwx------  1 andreochimal  staff  618 Nov  3 22:49 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorO/names_2003-05-04.dat
-rwx------  1 andreochimal  staff  627 Nov  3 22:49 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorO/names_2009-05-06.dat
-rwx------  1 andreochimal  staff  623 Nov  3 22:49 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorP/names_2017-05-17.dat
-rwx------  1 andreochimal  staff  621 Nov  3 22:49 /Users/andreochimal/Desktop/Compu/Resultado/CUAUHTEMOC/sectorP/names_2018-05-09.txt


Cambié los permisos de todos los archivos en los directorios *sectorO* y *sectorP* que tienen *-05-* en su nombre, lo cual indica que tienen mayo como fecha, y al final enlisté estos archivos para corroborar que el procedimiento fuera correcto.

### *find y grep*

In [193]:
find /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio -name "*.txt" | wc -l

    2036


In [242]:
find /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio -name "*.dat" | wc -l

    2122


Hay 2036 archivos con extensión *.txt* y 2122 archivos con extensión *.dat*.

In [23]:
grep -r Andreo /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio | wc -l

       0


In [33]:
grep -r Ernesto /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio | wc -l

       0


In [31]:
grep -r Andreo /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio/*/*/*2002* | wc -l

       0


No hay archivo existente con alguno de mis dos nombres. Así, mi nombre no aparece en los 4158 archivos. Mi nombre nunca coincide con mi año de nacimiento, pues no está en ningún archivo.

In [247]:
find /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio -type f | wc -l

    4158


In [246]:
find /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio -type d | wc -l

     188


Hay 188 directorios y 4158 archivos en esos directorios.

In [270]:
find /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio -type f -size -1150c | wc -l

    4158


Todos los 4158 archivos pesan menos de 1150 bytes.

In [253]:
find  /Users/andreochimal/Desktop/Compu/Computacion2020-8093/Practicas/directorio/*/sectorS -name "*2020*" | wc -l

       8


Hay 8 archivos en *sectorS* que tienen "2020" en su nombre.